test/DegenGambit.for.t.sol

```diff
@@ -0,0 +1,438 @@
+        vm.expectEmit();
+        emit Spin(player2, false);
+        degenGambit.spinFor{value: costToSpin}(player2, player2, false);
+
+        uint256 finalStreakDay = degenGambit.LastStreakDay(player2);
+        assertEq(finalStreakDay, block.timestamp / SECONDS_PER_DAY);
+
+        uint256 gambitSupplyFinal = degenGambit.totalSupply();
+        uint256 playerGambitBalanceFinal = degenGambit.balanceOf(player2);
+
+        assertEq(
+            gambitSupplyFinal,
+            gambitSupplyInitial + 2 * dailyStreakReward
+        );
+        assertEq(
+            playerGambitBalanceFinal,
+            playerGambitBalanceInitial + 2 * dailyStreakReward
+        );
+    }
+
+    function test_gambit_minted_on_streak_boosted_spinfor() public {
+        // Make sure the player has GAMBIT to boost with.
+        degenGambit.mint(player1, 2);
+
+        uint256 gambitSupplyInitial = degenGambit.totalSupply();
+        uint256 playerGambitBalanceInitial = degenGambit.balanceOf(player1);
+
+        uint256 dailyStreakReward = degenGambit.DailyStreakReward();
+
+        vm.startPrank(player1);
+
+        vm.expectEmit();
+        emit Spin(player2, true);
+        degenGambit.spinFor{value: costToSpin}(player2, player1, true);
+
+        uint256 gambitSupplyIntermediate = degenGambit.totalSupply();
+        uint256 playerGambitBalanceIntermediate = degenGambit.balanceOf(
+            player1
+        );
+
+        uint256 intermediateStreakDay = degenGambit.LastStreakDay(player1);
+        assertEq(intermediateStreakDay, block.timestamp / SECONDS_PER_DAY);
+
+        assertEq(gambitSupplyIntermediate, gambitSupplyInitial - 1);
+        assertEq(
+            playerGambitBalanceIntermediate,
+            playerGambitBalanceInitial - 1
+        );
+
+        vm.roll(block.number + 1);
+        // Tests the right end of the window for which the streak is active.
+        vm.warp(
+            (block.timestamp / SECONDS_PER_DAY) *
+                SECONDS_PER_DAY +
+                2 *
+                SECONDS_PER_DAY -
+                1
+        );
+
+        vm.expectEmit();
+        emit Transfer(address(0), player1, dailyStreakReward);
+        vm.expectEmit();
+        emit DailyStreak(player1, intermediateStreakDay + 1);
+        vm.expectEmit();
+        emit Spin(player2, true);
+        degenGambit.spinFor{value: costToSpin}(player2, player1, true);
+
+        uint256 finalStreakDay = degenGambit.LastStreakDay(player1);
+        assertEq(finalStreakDay, intermediateStreakDay + 1);
+
+        uint256 gambitSupplyFinal = degenGambit.totalSupply();
+        uint256 playerGambitBalanceFinal = degenGambit.balanceOf(player1);
+
+        assertEq(
+            gambitSupplyFinal,
+            gambitSupplyIntermediate + dailyStreakReward - 1
+        );
+        assertEq(
+            playerGambitBalanceFinal,
+            playerGambitBalanceIntermediate + dailyStreakReward - 1
+        );
+    }

```

Comment:
@mrk-hub : Can you add tests for weekly streaks on `spinFor`?

Can you also add tests for `spinFor` streaks in which the streak player differs from the spin player?